tests/test_list.cpp

```diff
@@ -71,6 +72,56 @@ class TestList : public CppUnit::TestFixture
     CPPUNIT_ASSERT_EQUAL(33, l2[2]);
   }
 
+  void bracedInit()
+  {
+    List<int> l1 {
+      1,
+      2,
+      3
+    };
+    CPPUNIT_ASSERT_EQUAL(3, static_cast<int>(l1.size()));
+    CPPUNIT_ASSERT_EQUAL(1, l1[0]);
+    CPPUNIT_ASSERT_EQUAL(2, l1[1]);
+    CPPUNIT_ASSERT_EQUAL(3, l1[2]);
+
+    List<int*> l2 {
+      new int(1),
+      new int(2),
+      new int(3)
+    };
+    l2.setAutoDelete(true);
+    CPPUNIT_ASSERT_EQUAL(3, static_cast<int>(l2.size()));
+    CPPUNIT_ASSERT_EQUAL(1, *l2[0]);
+    CPPUNIT_ASSERT_EQUAL(2, *l2[1]);
+    CPPUNIT_ASSERT_EQUAL(3, *l2[2]);
+
+    List<int> l3 = {
+      1,
+      2,
+      3
+    };
+    CPPUNIT_ASSERT_EQUAL(3, static_cast<int>(l3.size()));
+    CPPUNIT_ASSERT_EQUAL(1, l3[0]);
+    CPPUNIT_ASSERT_EQUAL(2, l3[1]);
+    CPPUNIT_ASSERT_EQUAL(3, l3[2]);
+
+    List<int*> l4 = {
+      new int(1),
```

Comment:
how nasty, I can't tell if this is a memory leak or not.

Reply:
The `ListPrivate` shared pointer is assigned a new `ListPrivate` object constructed from the braced list. The [assignment operator of the shared pointer](https://en.cppreference.com/w/cpp/memory/shared_ptr/operator%3D) releases ownership of the old object. When its reference count reaches 0 the object is deleted automatically.

So there shouldn't be any memory leak as long as autodelete is set on the `List`.

Reply:
I trust your judgement here.

In a different project, GCC's -fanalyzer complained when using shared_ptr(make_shared I think) and initializer_list saying there was a memory leak. I fixed by avoiding the latter and using std::move. No idea how much of a false positive it was.